daemon/daemon_unix.go

```diff
@@ -1184,3 +1194,30 @@ func setupOOMScoreAdj(score int) error {
 	f.Close()
 	return err
 }
+
+func (daemon *Daemon) initCgroupsPath(path string) error {
+	if path == "/" || path == "." {
+		return nil
+	}
+
+	daemon.initCgroupsPath(filepath.Dir(path))
+
+	root := "/sys/fs/cgroup/cpu"
+	path = filepath.Join(root, path)
+	sysinfo := sysinfo.New(false)
+	if err := os.MkdirAll(path, 0755); err != nil && !os.IsExist(err) {
+		return err
+	}
+	if sysinfo.CPURealTimePeriod && daemon.configStore.CPURealTimePeriod != 0 {
```

Comment:
Sorry, I should have clarified better. The issue here is not related to the container's cgroup specifically but the one Docker is configured to run in.

The issue this particular piece of code is meant to resolve is related to the `docker daemon --cpu-rt-runtime` not `docker run`

Discussed in this comment: https://github.com/docker/docker/pull/23430#issuecomment-234200112

I believe with it being code related to a `docker daemon` flag it should still reside here?